client/components/LearnerViewDirective.js

```diff
@@ -410,11 +469,21 @@ tie.directive('learnerView', [function() {
           loadQuestion(questionId, questionSet.getIntroductionParagraphs());
         };
 
-        $scope.submitCode = function(code) {
+        var callSolutionHandlerService = function() {
+          var code = this.code;
           SolutionHandlerService.processSolutionAsync(
-            prompts[currentPromptIndex], code,
-            question.getAuxiliaryCode(language), language
-          ).then(setFeedback);
+              prompts[currentPromptIndex], code,
+              question.getAuxiliaryCode(language), language
+              ).then(setFeedback);
+        };
+
+        $scope.submitCode = function(code) {
```

Comment:
Hmm... This is what I had originally, but I ran into some bug earlier that made me define a separate function and didn't revisit it. Thanks for the catch!